### DISCUSSION TITLE
part-ii/2020-102/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#128] [Question comments]
Part II 2020 Number Theory: Paper 1, Section I, H 

https://questions.tripos.org/part-ii/2020-102/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

A positive definite BQF $(a,b,c)$ is reduced if either $0 \leq b \leq a = c$ or $-a < b \leq a < c$.

To find the reduced BQFs with a given discriminant $d$, we use the result that $|b| \leq a \leq \sqrt{\frac{|d|}{3}}$ and $b \equiv d \pmod{2}$.

When $d = -20$, we have $\sqrt{\frac{|d|}{3}} < 3$, so since $b \equiv 0 \pmod{2}$, either $|b| = 0$ or $|b| = 2$.

If $|b| = 2$, then $a = 2$ also (by the inequality), which forces $c = 3$ (by solving $d = b^2 - 4ac$). This gives the reduced form $(2,2,3)$, i.e. $2x^2 + 2y^2 + 3y^2$ (recall that we cannot have $b = -a$).

If $|b| = 0$, then $a$ must be 0, 1 or 2. We have $-20 = -4ac$, so $ac = 5$. This clearly does not have an integer solution if $a = 0$ or $a = 2$, but if $a = 1$ we get the reduced form $(1,0,5)$ (i.e. $x^2 + 5y^2$).

Now we have a theorem that an integer $n$ is properly represented by some form of discriminant $d$ if and only if the congruence $u^2 \equiv d \pmod{4n}$ is soluble. Therefore if a prime $p \neq 5$ is represnted by $x^2 + 5y^2$, then $u^2 \equiv -20 \pmod{4p}$ is soluble (we can switch from "properly represented" to "represented" since we are dealing with primes coprime to the coefficients of the form). We know that $p=2$ is not represented by this form (we can list the smallest integers the form represents), so we can assume $(p,2) = 1$ and apply the Chinese Remainder Theorem. $u^2 \equiv -20 \equiv 0 \pmod{4}$ is certainly soluble, but we require that $u^2 \equiv - 20 \pmod{p}$ is soluble.

We have that $(-20/p) = (-5/p)$ (taking out a square factor), and $(-5/p) = (-1/p)(5/p) = (-1/p)(p/5)$ (using the law of quadratic reciprocity). By Euler's criterion, $(-1/p) = (-1)^\frac{p-1}{2}$, and $(p/5)$ can be evaluated by listing all the possibilities. 

We want that either $(p/5) = 1$ and $(-1/p) = 1$, so $p$ is 1 or 4 $\pmod{5}$ and $p \equiv 1 \pmod{4}$, which gives $p \equiv 1$ and $p \equiv 9 \pmod{20}$, or $(p/5) = -1$ and $(-1/p) = -1$, so $p$ is 2 or 3 $\pmod{5}$ and $p \equiv 3 \pmod{4}$, which gives $p \equiv 3$ and $p \equiv 7 \pmod{20}$.

</details>
